infrastructure/lib/stacks/s3.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+import {Stack, RemovalPolicy, StackProps} from "aws-cdk-lib";
+import { Bucket, BlockPublicAccess, BucketEncryption } from 'aws-cdk-lib/aws-s3';
+import { Construct } from "constructs";
+
+export class OpenSearchS3 extends Stack {
+
+    public readonly githubEventsBucket: Bucket;
+
+    constructor(scope: Construct, id: string, props?: StackProps) {
+        super(scope, id);
+
+        this.githubEventsBucket = new Bucket(this, 'OpenSearchMetricsGithubEvents', {
+            bucketName: "opensearch-project-github-events",
```

Comment:
Do we want to name the bucket? Usually it is not recommended to name the resources created via CDK. Let the CDK assign the random naming to avoid future failures.

Reply:
Just curious, why is it not recommended? We are indexing to this bucket in the automation app here: https://github.com/opensearch-project/automation-app/blob/main/src/call/github-events-to-s3.ts#L29

Reply:
s3 bucket names are unique globally (not just within the account but all accounts in AWS). When you name the resources, you cannot deploy multiple stacks using same CDK as it will throw an error with resource already exists. All the resources in CDK have a resource identifier or logical name but adding a resource name might not be a good idea. You can deploy the stack and use the name created by the CDK in the automation app url mentioned above.

Reply:
Sounds good, removed the bucket name and changed the logical ID to be more specific

Reply:
@bshien you can add the bucket name to the secret and use the secret here https://github.com/opensearch-project/automation-app/blob/main/src/call/github-events-to-s3.ts#L29.

Reply:
Other option we can go is allow the `bucketName` to passed as a stack props, then its upto the user to pass the bucket name.

Reply:
Check out this PR here: https://github.com/opensearch-project/automation-app/pull/26

Reply:
Just added one comment here https://github.com/opensearch-project/automation-app/pull/26/files#r1793853695, in that case just create the bucket and update the secret. Thanks